js/packages/openinference-instrumentation-llama-index/src/utils.ts

```diff
@@ -134,23 +154,83 @@ function getDocumentAttributes(
  * and constructs an Attributes object with the relevant semantic conventions
  * for embeddings.
  *
- * @param {Object} embedInfo - The embedding information.
- * @param {string} embedInfo.input - The input text for the embedding.
- * @param {number[]} embedInfo.output - The output embedding vector.
+ * @param {Object} embeddingInfo - The embedding information.
+ * @param {string} embeddingInfo.input - The input text for the embedding.
+ * @param {number[]} embeddingInfo.output - The output embedding vector.
  * @returns {Attributes} The constructed embedding attributes.
  */
-function getQueryEmbeddingAttributes(embedInfo: {
+function getQueryEmbeddingAttributes(embeddingInfo: {
   input: string;
   output: number[];
 }): Attributes {
   return {
     [`${SemanticConventions.EMBEDDING_EMBEDDINGS}.0.${SemanticConventions.EMBEDDING_TEXT}`]:
-      embedInfo.input,
+      embeddingInfo.input,
     [`${SemanticConventions.EMBEDDING_EMBEDDINGS}.0.${SemanticConventions.EMBEDDING_VECTOR}`]:
-      embedInfo.output,
+      embeddingInfo.output,
   };
 }
 
+function getLLMChatAttributes(chatInfo: {
+  input: llamaindex.LLMChatParamsNonStreaming<object, object>;
+  output: llamaindex.ChatResponse<object>;
+}) {
+  const LLMAttr: Attributes = {};
+
+  chatInfo.input.messages.forEach((msg, idx) => {
+    const inputPrefix = `${SemanticConventions.LLM_INPUT_MESSAGES}.${idx}`;
+    LLMAttr[`${inputPrefix}.${SemanticConventions.MESSAGE_ROLE}`] =
+      msg.role.toString();
```

Comment:
what are these types that require you to call toString() on them?

Reply:
They are ChatMessage objects: https://ts.llamaindex.ai/api/type-aliases/ChatMessage

Reply:
gotcha that makes sense!